doc/reference/networking/openthread_integration.rst

```diff
@@ -0,0 +1,172 @@
+   :ref:`k_fifo <fifos_v2>`). By default there is only one receive queue in
+   the system, but it is possible to have up to 8 receive queues.
+   These queues will process incoming packets with different priority.
+   See :ref:`traffic-class-support` for more details. The receive queues also
+   act as a way to separate the data processing pipeline (bottom-half) as
+   the device driver is running in an interrupt context and it must do its
+   processing as fast as possible.
+
+3. The 802154 RX Radio driver thread does the upper-half processing of the
+   received IEEE 802.15.4 radio frame. As a result it puts a work item
+   with the net_recv_data() to have the frame processed.
+
+4. The work queue thread rx_workq calls the registered handler for every queued frame.
+   In this case the registered handler openthread_recv() checks if the frame is of the 
+   IEEE 802.15.4 type and if it is the case it insertd the frame in the rx_pkt_fifo and returns the NET_OK.
+
+5. The OpenThread thread gets a frame from the FIFO and processes it.
+   It also handles IP header compression, reassembly of fragmented traffic.
+
+6. As soon as it detects a valid IPv6 packet that needs to be handled by the 
+   higher layer it calls the registered callback ot_receive_handler()
+   which creates a buffer for a net_pkt that is going to be passed to the Zephyr's IP stack
+   and calls the net_recv_data() to have it processed.
+
+7. This time the openthread_recv() called by the work queue returns NET_CONTINUE
+   indicating that the valid IPv6 packet is present and needs to be processed by
+   the Zephyr's higher layer.
+
+8. The net_ipv6_input() passes the packet the next higher layer.
+
+9. The packet is then passed to L3 processing. If the packet is IP based,
+   then the L3 layer processes the IPv6.
+
+10. A socket handler then finds an active socket to which the network packet
+   belongs and puts it in a queue for that socket, in order to separate the
+   networking code from the application. Typically the application is run in
+   userspace context and the network stack is run in kernel context.
+
+11. The application will then receive the data and can process it as needed.
+   The application should have used the
+   :ref:`BSD socket API <bsd_sockets_interface>` to create a socket
+   that will receive the data.
+
+TX path
+*******
+
+.. figure:: zephyr_netstack_openthread-tx_sequence.svg
+    :alt: OpenThread Application TX data flow
+    :figclass: align-center
+
+    OpenThread Application TX data flow
+
+Data transmitting (TX)
+
+
+1. The application uses the
+   :ref:`BSD socket API <bsd_sockets_interface>` when sending the data.
+   However, direct interaction with the OpenThread API is possible - e.g.
+   to utilize it's CoAP implementation.
+
+2. The application data is prepared for sending to kernel space and then
+   copied to internal net_buf structures.
+
+3. Depending on the socket type, a protocol header is added in front of the
```

Comment:
Is IPv6 not involved here (as is in RX)?

Reply:
It seems that the UDP is injected to the OT directly.